src/rotation_error.jl

```diff
@@ -0,0 +1,95 @@
+
+"""
+    RotationError{T<:Real, D<:ErrorMap} <: StaticVector{3,T}
+
+A three-parameter rotation error, converted to/from a `UnitQuaternion` using the
+`ErrorMap` `D`.
+
+# Usage
+A `RotationError` is typically created using one of the following methods
+
+    rotation_error(R1::Rotation, R2::Rotation, error_map::ErrorMap)
+    R1 ⊖ R2
+
+which compute the difference between the rotations `R1` and `R2` and convert the
+result to a 3D rotation error using `error_map`.
+
+The error can be "added" back to a rotation using the inverse operation:
+
+    add_error(R1::Rotation, e::RotationError)
+    R1::Rotation ⊕ e::RotationError
+"""
+struct RotationError{T,D} <: StaticVector{3,T}
+    err::SVector{3,T}
+    map::D
+    @inline function RotationError(err::SVector{3,T}, map::D) where {T,D <: ErrorMap}
+        new{T,D}(err, map)
+    end
+end
+
+"""
+    inverse_map(e::RotationError)
+
+Convert the error back to a `UnitQuaternion` using the error map in `e`
+""" # QUESTION: maybe just use `UnitQuaternion`?
+@inline forward_map(e::RotationError)::Rotation = inverse_map(e.map, e.err)
```

Comment:
This is confusing - is this `forward_map` or inverse map.  The docstring says inverse map, the function name is forward_map, and the implementation uses inverse_map!

Reply:
Fixed. Changed to `UnitQuaternion` since it's really just converting the `RotationError` to a `UnitQuaternion`.